aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -284,6 +286,14 @@ def __init__(
             env=os.getenv(constants.EVENT_HANDLER_DEBUG_ENV, "false"), choice=debug
         )
 
+        if serializer:
+            self._serializer = serializer
+        elif self._debug:
+            """Does a concise json serialization or pretty print when in debug mode"""
+            self._serializer = partial(json.dumps, indent=4, cls=Encoder)
+        else:
+            self._serializer = partial(json.dumps, separators=(",", ":"), cls=Encoder)
```

Comment:
```suggestion
        self._serializer = serializer or partial(json.dumps, separators=(",", ":"), cls=Encoder)

        if self._debug:
            """Does a concise json serialization or pretty print when in debug mode"""
            self._serializer = partial(json.dumps, indent=4, cls=Encoder)
```

Syntactic sugar, since self._debug will almost always be False.